src/domain/rate-limit/index.ts

```diff
@@ -1,11 +1,76 @@
+import {
+  getFailedLoginAttemptPerIpLimits,
+  getFailedLoginAttemptPerPhoneLimits,
+  getInvoiceCreateAttemptLimits,
+  getInvoiceCreateForRecipientAttemptLimits,
+  getOnChainAddressCreateAttemptLimits,
+  getRequestPhoneCodePerIpLimits,
+  getRequestPhoneCodePerPhoneLimits,
+  getRequestPhoneCodePerPhoneMinIntervalLimits,
+} from "@config/app"
+
+import {
+  InvoiceCreateForRecipientRateLimiterExceededError,
+  InvoiceCreateRateLimiterExceededError,
+  OnChainAddressCreateRateLimiterExceededError,
+  UserLoginIpRateLimiterExceededError,
+  UserLoginPhoneRateLimiterExceededError,
+  UserPhoneCodeAttemptIpRateLimiterExceededError,
+  UserPhoneCodeAttemptPhoneMinIntervalRateLimiterExceededError,
+  UserPhoneCodeAttemptPhoneRateLimiterExceededError,
+} from "./errors"
+
 export const RateLimitPrefix = {
   requestPhoneCodeAttemptPerPhone: "phone_code_attempt_phone_code",
   requestPhoneCodeAttemptPerPhoneMinInterval:
     "phone_code_attempt_phone_code_min_interval",
   requestPhoneCodeAttemptPerIp: "phone_code_attempt_ip",
-  failedLoginAttemptPerIp: "login_attempt_ip",
   failedLoginAttemptPerPhone: "login_attempt_phone",
+  failedLoginAttemptPerIp: "login_attempt_ip",
   invoiceCreate: "invoice_create",
   invoiceCreateForRecipient: "invoice_create_for_recipient",
   onChainAddressCreate: "onchain_address_create",
 } as const
+
+export const RateLimitConfig: { [key: string]: RateLimitConfig } = {
+  requestPhoneCodeAttemptPerPhone: {
+    key: RateLimitPrefix.requestPhoneCodeAttemptPerPhone,
+    limits: getRequestPhoneCodePerPhoneLimits(),
+    error: UserPhoneCodeAttemptPhoneRateLimiterExceededError,
+  },
+  requestPhoneCodeAttemptPerPhoneMinInterval: {
+    key: RateLimitPrefix.requestPhoneCodeAttemptPerPhoneMinInterval,
+    limits: getRequestPhoneCodePerPhoneMinIntervalLimits(),
+    error: UserPhoneCodeAttemptPhoneMinIntervalRateLimiterExceededError,
+  },
+  requestPhoneCodeAttemptPerIp: {
+    key: RateLimitPrefix.requestPhoneCodeAttemptPerIp,
+    limits: getRequestPhoneCodePerIpLimits(),
+    error: UserPhoneCodeAttemptIpRateLimiterExceededError,
+  },
+  failedLoginAttemptPerPhone: {
+    key: RateLimitPrefix.failedLoginAttemptPerPhone,
+    limits: getFailedLoginAttemptPerPhoneLimits(),
+    error: UserLoginPhoneRateLimiterExceededError,
+  },
+  failedLoginAttemptPerIp: {
+    key: RateLimitPrefix.failedLoginAttemptPerIp,
+    limits: getFailedLoginAttemptPerIpLimits(),
+    error: UserLoginIpRateLimiterExceededError,
+  },
+  invoiceCreate: {
```

Comment:
I have renamed this to `invoiceCreateForSelf` in this pending draft PR: https://github.com/GaloyMoney/galoy/pull/936

Reply:
Should I wait until #936 is reviewed/merged to then merge this one?

Reply:
no you can merge it. it's more FI. #936 has some dependancy on #921 also.